### PR TITLE
Merge README instructions into bootloader README

### PR DESCRIPTION
**Pull request by AGlass0fMilk**
This commit removes most of the instructions from this repository's README. The removed instructions have been reworded and merged into the corresponding bootloader repository's README.

This commit also introduces use of the new `boot_get_current_version` function in mcuboot (not yet merged).

This commit temporarily retargets the mcuboot.lib dependency to a test branch since some required changes have not yet been merged into mcuboot master.

@LDong-Arm

### CODE REVIEW

**Comment by AGlass0fMilk** [comment]
See corresponding PR for bootloader: AGlass0fMilk/mbed-mcuboot-demo#6

File: main.cpp [hunk 1]
```diff
@@ -43,7 +42,16 @@ int main()
...
-    tr_info("Hello version %s", version);
+    // Get the current version from the mcuboot header information
+    struct image_version version;
+    ret = boot_get_current_version(&version);
+    if(ret == 0) {
+        tr_info("Hello version %d.%d.%d+%lu", version.iv_major, version.iv_minor,
+                                              version.iv_revision, version.iv_build_num);
+    } else {
+        tr_error("Failed to load version information: %d", ret);
+    }
```
Comment (LDong-Arm) [inline main.cpp:53]: Good idea!

**Review by LDong-Arm — APPROVED** [review]

File: main.cpp (hunk 1 above)
Comment (AGlass0fMilk) [reply main.cpp:53]: Yes, I had to add that API to mcuboot... see mcu-tools/mcuboot#863

I'm hoping they accept the new function into the API.

**Comment by AGlass0fMilk** [comment]
@LDong-Arm Thanks for your work on the demo, it makes it much more educational by having the button presses at each stage. I think the new demo is much more explanatory and straightforward.

You can try out the hex editing commands I put in the walkthrough. I think it's a pretty simple and clever way to simulate an update.

**Comment by LDong-Arm** [comment]
@AGlass0fMilk You're welcome.

Extra tip which we tried and you can integrate:

```c
    // Check if an update has been performed
    int swap_type = boot_swap_type();
    int ret;
    switch(swap_type) {
        case BOOT_SWAP_TYPE_NONE:
            tr_debug("Regular boot");
            break;

        case BOOT_SWAP_TYPE_REVERT:
            // After MCUboot has swapped a (non-permanent) update image
            // into the primary slot, it defaults to reverting the image on the NEXT boot.
            // This is why we see "[INFO][MCUb]: Swap type: revert" which can be misleading.
            // Confirming the CURRENT boot dismisses the reverting.

            // PLACEHOLDER: Do some self tests
            tr_info("Firmware updated successfully");
            tr_info("Press the button to confirm, or reboot to revert the update");
            while(gpio_read(&btn) != 0); // wait for button press
            while(gpio_read(&btn) == 0); // wait for button release

            ret = boot_set_confirmed();
            if (ret == 0) {
                tr_info("Current firmware set as confirmed");
            } else {
                tr_error("Failed to confirm the firmware: %d", ret);
            }
            break;

        // Note: Below are intermediate states of MCUboot and
        // should never reach the application...
        case BOOT_SWAP_TYPE_FAIL:   // Unable to boot due to invalid image
        case BOOT_SWAP_TYPE_PERM:   // Permanent update requested (when signing the image) and to be performed
        case BOOT_SWAP_TYPE_TEST:   // Revertable update requested and to be performed
        case BOOT_SWAP_TYPE_PANIC:  // Unrecoverable error
        default:
            tr_error("Unexpected swap type: %d", swap_type);
    }
```

Note: It's better to check a button is both pressed and released. Without waiting for the release, if the code between two button presses are super quick to run, your press (which takes time) would trigger both...

**Pull request merged**